### DISCUSSION TITLE
Automation manager policy checks - when do they run?

### DISCUSSION THREAD

**Discussion by ghost** [amidaware/tacticalrmm#1493] [Q&A]
At first connect? Hourly?

**Comment by silversword411** [comment]
At the time interval you've specified.

Agent checks every 2mins by default if you haven't specified something else. You have server check preference. Agent check preference and check preference as well that can be adjusted.

If you're asking when the agent gets pushed the policy configuration change it's pretty quick,

**Reply by ghost** [reply]
Where are these pref settings? I'll be damned if I can find them!

**Comment by silversword411** [comment]
Agent setting

![2023-04-25_090633 - agent check interval](https://user-images.githubusercontent.com/13395348/234286459-69012912-8246-4dfa-aa3c-f6b64b47a758.png)

Script setting
![2023-04-25_090633 - script check interval](https://user-images.githubusercontent.com/13395348/234286524-bd58aebf-6d22-4060-9f7e-5fdafd56e6ea.png)

**Reply by ghost** [reply]
So you can only edit on a per-agent basis, not globally?

**Comment by silversword411** [comment]
Correct.

Just set it on each script. 

There's no management command atm to adjust default agent script times atm. But just because the agent is "checking" every 120 seconds to see IF there is a script run, doesn't make any scripts run unless it's time has come (by not specifying a default time).
